nanodbc/nanodbc.cpp

```diff
@@ -3429,14 +3429,15 @@ void result::result_impl::get_ref_from_string_column(short column, T& result) co
 }
 
 template <typename T>
-T* result::result_impl::ensure_pdata(short column) const
+std::unique_ptr<T, std::function<void(T*)>> result::result_impl::ensure_pdata(short column) const
 {
     bound_column& col = bound_columns_[column];
     SQLLEN ValueLenOrInd;
     SQLRETURN rc;
     if (is_bound(column))
     {
-        return (T*)(col.pdata_ + rowset_position_ * col.clen_);
+        return std::unique_ptr<T, std::function<void(T*)>>(
+            (T*)(col.pdata_ + rowset_position_ * col.clen_), [](T* ptr) {});
```

Comment:
I believe this needs a comment on why no-op deleter here and why the default deleter in the other place.

Reply:
Thanks.

Added a few notes - let me know if this is what you had in mind.

Reply:
LGTM